src/interactivity/base/InteractivityFactory.cpp

```diff
@@ -499,7 +499,9 @@ void InteractivityFactory::_WritePseudoWindowCallback(bool showOrHide)
     auto& gci = ServiceLocator::LocateGlobals().getConsoleInformation();
     if (const auto io = gci.GetVtIo())
     {
-        io->SetWindowVisibility(showOrHide);
+        char buf[] = "\x1b[1t";
+        buf[2] = showOrHide ? '1' : '2';
+        io->WriteUTF8(buf);

```

Comment:
This is a problem when the terminal is in VT52 mode. You can see what happens by running vttest, and selecting "Test of VT52 mode". Then when you're viewing the first test screen, click on the taskbar icon to minimize and restore the terminal. Notice the `2t1t` characters being written to the screen.

My initial thought was that this might be one of those cases where we discussed resetting the terminal to a neutral state when a console API was used. However, I think it's more complicated than that. Because in this case we don't actually want to leave VT52 mode permanently, so we also need to restore the mode afterwards. Maybe it's better if we just skip sending these show/hide sequences when we're in VT52 mode?

Although now that I think about it, there may be other situations where this could break as well. Like if we're in the middle of a long running `DCS` sequence, this could end up being injected in the middle, which would terminate the sequence, and the remaining content would end up output to the screen. This might already be a problem with the existing conpty implementation though. If so, maybe we can ignore it for now.

Reply:
> This might already be a problem with the existing conpty implementation though. If so, maybe we can ignore it for now.

I can properly test it later, but I strongly suspect it's a problem already. We write DCS directly into the `VtEngine` buffer via `WriteTerminalW` and force-flush it. `SetWindowVisibility` does the exact same thing. This should mean that they can end up being interleaved in the output pipe.

Reply:
Btw, I'm personally quite happy to leave all of these edge cases for later. I'm just noting whatever issues I find so you're aware of them.